mlrun/api/api/utils.py

```diff
@@ -306,6 +306,42 @@ def delete_notification_params_secret(
     )
 
 
+def validate_and_mask_notification_list(
+    notifications: typing.List[
+        typing.Union[mlrun.model.Notification, mlrun.common.schemas.Notification, dict]
+    ],
+    parent: str,
+    project: str,
+) -> typing.List[mlrun.model.Notification]:
+    notification_objects = []
+
+    for notification in notifications:
+        if isinstance(notification, dict):
+            notification_object = mlrun.model.Notification.from_dict(notification)
+        elif isinstance(notification, mlrun.common.schemas.Notification):
+            notification_object = mlrun.model.Notification.from_dict(
+                notification.dict()
+            )
+        elif isinstance(notification, mlrun.model.Notification):
+            notification_object = notification
+        else:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                "notification must be a dict or a Notification object"
+            )
+
+        # validate notification schema
+        mlrun.common.schemas.Notification(**notification_object.to_dict())
```

Comment:
will this raise an exception if it does not adhere to the schema?
If so do we catch it or let it go 💥 ?

Reply:
it does raise and we do let it go boom :) 
It will cause the API to raise a Bad request on schema validation